src/main/java/com/mixpanel/android/util/RemoteService.java

```diff
@@ -10,7 +10,7 @@
 
 
 public interface RemoteService {
-    boolean isOnline(Context context);
+    boolean isOnline(Context context, OfflineMode offlineMode);
```

Comment:
As conclusion. I do not have a strong opinion about enforce `OfflineMode` on the interface or use it directly by the implementors. But I would recommend that for consistency `SSLSocketFactory` should use the same approach: or both are present on the interface, or both should be retrieved via `MPConfig`

@patedit what do you think ?